.github/workflows/updateReadme.yml

```diff
@@ -0,0 +1,53 @@
+name: 'update-readme'
+
+on:
+  # cron schedule runs every Tuesday at 5pm UTC following public release of new cloud sdk version
+  schedule:
+    -cron: '0 17 * * 2'
+
+jobs:
+  build:
+    permissions:
+      contents: 'write'
+      id-token: 'write'
+      pull-requests: write
+    runs-on: 'ubuntu-latest'
+
+    steps:
+    - uses: 'actions/checkout@v3'
+
+    - uses: 'actions/setup-node@v3'
+      with:
+        node-version: '16.x'
+    
+    # Used to authenticate setup-gcloud in update script
+    - name: 'Auth with WIF'
+      uses: google-github-actions/auth@v0
+      with:
+        workload_identity_provider: # TODO replace with WIF
+        service_account: # TODO: replace with SA
+    
+    - name: 'Build & Run Update Readme Script'
```

Comment:
I might be missing something, but this seems a little complex to me. I think we could have three steps:

1. `uses: google-github-actions/setup-gcloud`
2. `run: gcloud --format=json functions event-types list > events.json`
3. render template with events.json as input